### ISSUE TITLE
Progress Report

### ISSUE THREAD

**Issue by mycotek** [mycotek/Proposal#3]
@jelmerp

**Comment by jelmerp** [comment]
Hi Nicolle,

This looks very good overall!

| Aspect              | Max. points | Your points |                                                                                                |
| ------------------- | :---------: | :---------: | ---------------------------------------------------------------------------------------------- |
| Scripts             |      4      |      4      | Great job!                                                                                     |
| Overview of scripts |      3      |     2.5     | `Overview.md` is not entirely clear, the to-do list actually gives better overview of scripts. |
| To-do list          |     1.5     |     0.5     | It seems like you have already done what's in your to-do list? <sup>[1]</sup>                  |
| Git commits         |     1.5     |      1      | Try to have smaller commits <sup>[2]</sup>.                                                    |
| **TOTAL**           |   **10**    |    **8**    |                                                                                                |

<sup>[1]</sup> I assume you want to further improve these scripts in some ways (?), but that's what you should have specified in the to-do list.
<sup>[2]</sup> E.g. the commit with message _"Updates made to main.sh, doBlast.sh, and created getFasta.py for downloading fasta files from NCBI"_ should have been at least 3 separate commits.

### Main comments

- Your scripts are all very good! For instance, I like that you have a setup script and in it create a Conda environment if it doesn't already exist. Your `doBlast.sh` script is also very well-structured, with several appropriate tests, and so on (but see the next point). And finally, it is great that you already have a script `main.sh` that ties everything together!

- From a workflow perspective, it would be better to split `doBlast.sh` into two scripts: one that runs BLAST, and one that generates a tree. It is often easier to keep adding to the same script, as you mentioned in your to-do list, since you already have the boilerplate code and the input for the next step is right there. But in the long run, your workflow will be clearer (e.g., now, a script called `doBlast.sh` that in fact also builds a tree...), easier to adapt and easier to rerun when you split it up into small, logical units. That said, the script is still not particularly long even with these two tasks combined, so it is not a particularly big deal.

- It would be good to improve some of the documentation of what the workflow does as a whole, primarily the `Overview.md` document. I had to do some puzzling to find out what's going on and am still confused about one aspect (see below). For example, in the `Overview.md`, you only mention one script by name (`setup.sh`) and you have a list of three, while you currently have 4 scripts (`getFasta.py` is not described there). Then, at the bottom, you say "The final script will simply run the first 4 scripts in sequence." -- but that should be "3 scripts", right?

- Your scripts are fairly well documented (commented), but some additional details would be good. For instance, `main.sh` takes an argument which is called `the_input`, and it is not sufficiently clear what kind of input that would be. Only after looking at the `test.sh` script did I realize that your input is the Latin name of a species.

- I am still confused about the identity of the FASTA file(s) that will be blasted. It is not really explicitly mentioned anywhere, though it seems implied that you are processing a specific gene. However, from looking at the scripts, it looks like you are more or less randomly selecting a FASTA file returned by the NCBI nucleotide database for a given organism? Why are you doing this? Is it just a proof-of-principle that you would, for real analyses, modify to also search for a gene name? For this final project, it is fine that your workflow does something like merely randomly selecting a gene, but you should make it clear that it does so.

- **Please use (more) subdirectories, most of all you need one for the scripts.** You also have a few testing files -- I would put those in a dir like `sandbox`.

### Minor comments

- Use "strict" Bash settings (`set -euo pipefail`). In `doBlasth.sh`, you do have them, but they are commented out.

- Your `doBlast.sh` contains commented-out `#SBATCH` lines. One thing to note is that if you don't actually submit a script to the SLURM queue, `#SBATCH` lines would just be interpreted as comments (after all, they are special comment lines). Therefore, if you want to keep those lines but don't always want to submit to the queue, you can keep them in the script as-is and don't need to (further) comment them out.

- This is very minor, but instead of having the main part of your `doBlasth.sh` inside an `if`-statement, I would "turn the `if`-statement around": if the extension is not `fasta` or `fna`, then exit the script with `exit 1`. After such a statement, you can continue with the script outside of the `if`-statement: if the extension would have been wrong, the script would have already exited. (And I would take an equivalent strategy with the tree-creation that first tests if there is output).

- On line 39 in `doBlast.sh`, you do `fileName=$(basename "$blasting")` but you had already assigned that to a variable on line 13: `blastingFileName=$(basename "$blasting")`.
